### DISCUSSION TITLE
want to define a new boundary three layers above the boundary exists

### DISCUSSION THREAD

**Discussion by ziyiwang141319** [idaholab/moose#26167] [Q&A Meshing] [answered]
I want to define a new boundary several layers (2-5 as i define) above the boundary exists. I am thinking of use NearestNodeLocator and BoundingBoxNodeSystem, but I am kind of unclear how to reach the goal. or is there a better way? And for NearestNodeLocator and BoundingBoxNodeSystem, can I use the mesh from a file? (It did not work when I tried it). I am not very familar with generate mesh using Moose, I read a disscussion mentioned scratch in moose. Where can I activate it? Thanks!!

**Comment by GiudGiud** [comment, marked as answer]
Hello

We have not done much on boundary layers afaik
To do this, I would create a BoundaryLayerExtraIDGenerator that sets an extra element integer equal to the boundary layer index.

this new mesh generator, that only sets an extra element id, could then act on moose-generated meshes or external meshes loaded using a FileMeshGenerator the same way

Guillaume

**Reply by ziyiwang141319** [reply]
Thanks a lot. I will try.
